lib/adapter.js

```diff
@@ -0,0 +1,494 @@
+                messageMetrics.sent = 1;
+            }
+			
+		} else if( types[data.type] == 'receive' ) {
+			if (messageMetrics.received != undefined) {
+                messageMetrics.received++;
+            } else {
+                messageMetrics.received = 1;
+            }
+		}
+	}
+	amapi.on('strong-mq', incrementCounts);
+	amapi.on('axon', incrementCounts);
+}
+
+function createObjectListener() {
+	amapi.on('objectType', function(objectType) {
+
+	});
+	amapi.on('objectSize', function (objectSize) {
+		
+	});
+}
+
+function createProbeListener(probeName){
+	amapi.on(probeName, function(res){
+		var duration = res.duration;
+		var currentMetrics;
+		
+		// If no metrics object exists, create one
+		if (probeMetrics[probeName] == undefined){
+			probeMetrics[probeName] = {count: 0, runningTotal: 0};
+		}
+		
+		// Update all metrics accordingly
+		updateMetrics(probeName, duration); 
+	});
+}
+
+function getEventLoopData() {
+	// Calculate average
+	var count = eventLoopMetrics.count;
+	var average = eventLoopMetrics.averageRunningTotal / count;
+
+	// Build JSON
+	var eventLoopEvent = {
+		"count" : count,
+		"minimum" : eventLoopMetrics.min,
+		"maximum" : eventLoopMetrics.max,
+		"average" : average
+	};
+
+	// Reset values
+	eventLoopMetrics.count = 0;
+	eventLoopMetrics.min = null;
+	eventLoopMetrics.max = null;
+	eventLoopMetrics.averageRunningTotal = 0;
+
+	return eventLoopEvent
+}
+
+function getCpuData() {
+	var system = cpuMetrics.system / cpuMetrics.count;
+	var user = cpuMetrics.user / cpuMetrics.count;
+	var cpuEvent = {
+		"system" : system,
+		"user" : user,
+		"total" : system + user
+	};
+
+	// Reset values
+	cpuMetrics.system = 0;
+	cpuMetrics.user = 0;
+	cpuMetrics.count = 0;
+
+	return cpuEvent;
+}
+
+function getHTTPData() {
+	var httpEvent = {
+		"count" : httpMetrics.count
+	}
+
+	// Reset Count
+	httpMetrics.count = 0;
+
+    return httpEvent
+}
+
+function getProbeData(probeName){
+	
+	//Get data for given probe
+	var probeData = probeMetrics[probeName];
+	var count = probeData.count;
+	var average;
+	
+	if (count != 0){
+		average = probeData.runningTotal / count;
+	} else average = 0; 
+	
+	var probeEvent = {
+		"count" : count,
+		"minimum" : probeData.min,
+		"maximum" : probeData.max,
+		"average" : average
+	}
+	
+	// Reset values
+	probeMetrics[probeName] = undefined;
+	
+	return probeEvent;
+}
+
+function getHeapData() {
+	var jsonToReturn = heapMetrics;
+
+	// Reset JSON
+	heapMetrics = {};
+
+	return jsonToReturn
+}
+
+function getGCHeapData() {
+	if (gcMetrics.count > 0) {
+		var usedHeap = gcMetrics.used / gcMetrics.count;
+	
+		//Reset values
+		gcMetrics.used = 0;
+		gcMetrics.count = 0;
+	} else {
+		usedHeap = -1;
+	}
+	return usedHeap;
+}
+
+function getMessageData() {
+    if (messageMetrics.sent != undefined && messageMetrics.received != undefined) {
+        var oldMessageMetrics = messageMetrics;
+        messageMetrics = {};
+        return oldMessageMetrics;
+    }
+    return null;	
```

Comment:
redundant line, functions return `undefined` already (unless you need to distinguish between null and undefined, but I don't think you do)